### ISSUE TITLE
Protect all code blocks with `norust`

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-rust#92]
We need to parse the CommonMark comments and protect all code blocks with `norust`.

## Background

The Protobuf and OpenAPI v3 documentation comments are in Markdown format, they are supposed to be ([aip.dev/192](https://google.aip.dev/192)) in [CommonMark](https://commonmark.org/) format, though sometimes they deviate from the AIP.

The generator converts these Protobuf comments into Rustdoc comments.

Rust tries to compile all code blocks unless protected by a `norust` mark. 

Block quotes appear in many [different forms](https://spec.commonmark.org/0.31.2/#block-quotes) in the CommonMark spec.

In #57 I fixed fenced code blocks (that is, those protected by backticks `` ``` ``), then I thought of other ways to specify code quotes (indented multiple spaces).

## Overview

We need to parse the CommonMark comments, apply changes in the syntax tree, and re-emit Markdown.  Probably in the Codec's `FormatDocComments()` function.  As a starting point, consider this Golang library: https://github.com/yuin/goldmark

## Alternatives Considered

We could try to perform this cleanup using heuristics. Shudder. 😨

**Comment by coryan** [comment]
I think the hacks in #346 are enough to make this issue non-blocking for a release.

**Comment by coryan** [comment]
There is a code block where we could do better:

https://docs.rs/gcp-sdk-type/0.1.0-rc2/gcp_sdk_type/model/struct.PhoneNumber.html

**Comment by mpeddada1** [comment]
@coryan Yup! I think this can be closed now that https://github.com/googleapis/google-cloud-rust/pull/693 is merged.
